### DISCUSSION TITLE
set Customized session properties

### DISCUSSION THREAD

**Discussion by cabbage18** [trinodb/trino#21976] [Q&A]
Does Trino provide the capability to set Customized session properties before executing SQL queries, which can be utilized by Trino SPI plugins to perform customized processing based on the property values?

**Comment by anilsomisetty** [comment]
When we run the query "show session" by connecting to trino server it shows all the available session properties which can be set during query execution, are you looking for this @cabbage18 ?

**Comment by jack-o-hara** [comment]
The ability to set a custom named session property would also be useful for Gateway routing purposes.
